docs/resources/kps_keypair.md

```diff
@@ -0,0 +1,131 @@
+---
+subcategory: "Data Encryption Workshop (DEW)"
+---
+
+# huaweicloud_kps_keypair
+
+Manages a keypair resource within HuaweiCloud.
+
+By default, key pairs use the SSH-2 (RSA, 2048) algorithm for encryption and decryption.
+
+Keys imported support the following cryptographic algorithms:
+
+ * RSA-1024
+ * RSA-2048
+ * RSA-4096
+
+## Example Usage
+
+### Create a new keypair which scope is Tenant-level and the private key is managed by HuaweiCloud
+
+```hcl
+resource "huaweicloud_kms_key" "test" {
+  key_alias = "kms_test"
+}
+
+resource "huaweicloud_kps_keypair" "test-keypair" {
+  name                    = "my-keypair"
+  type                    = "ssh"
+  scope                   = "domain"
+  encryption_type         = "kms"
+  encryption_kms_key_name = huaweicloud_kms_key.test.key_alias
+}
+```
+
+### Create a new keypair and export private key to current folder
+
+```hcl
+resource "huaweicloud_kps_keypair" "test-keypair" {
+  name     = "my-keypair"
+  key_file = "private_key.pem"
+}
+```
+
+### Import an exist keypair
+
+```hcl
+resource "huaweicloud_kps_keypair" "test-keypair" {
+  name       = "my-keypair"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAlJq5Pu+eizhou7nFFDxXofr2ySF8k/yuA9OnJdVF9Fbf85Z59CWNZBvcAT... root@terra-dev"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the keypair resource. If omitted, the
+  provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies a unique name for the keypair. The name can contain a maximum of 64
+ characters, including letters, digits, underscores (_), and hyphens (-).
```

Comment:
underscores (_), and hyphens (-).
remove the comma before 'and'

Reply:
done